xapian-letor/configure.ac

```diff
@@ -377,13 +379,279 @@ dnl this warning.
 PERCENT='%'
 AC_SUBST([PERCENT])
 
+dnl If valgrind is installed and new enough, we use it for leak checking in the
+dnl testsuite.  If VALGRIND is set to an empty value, then skip the check and
+dnl don't use valgrind.
+if test -n "${VALGRIND-unset}" ; then
+  AC_PATH_PROG([VALGRIND], [valgrind], [])
+  if test -n "$VALGRIND" ; then
+    dnl Check that the installed valgrind version works, and supports the
+    dnl options we use.  This means we won't try to use valgrind < 3.3.0
+    dnl (released 7/12/2007) since before that --log-file didn't expand
+    dnl %p (and appended the process id).
+    dnl
+    dnl No need to check for VALGRIND_COUNT_LEAKS now - that was added before
+    dnl 2.0.0.
+    AC_MSG_CHECKING([if valgrind supports --log-file with %p])
+    vglog=config.vglog.%p.tmp
+    vglogwild="config.vglog.*.tmp*"
+    rm -f $vglogwild
+    if $VALGRIND --log-file="$vglog" -q true 2>&AS_MESSAGE_LOG_FD ; then
+      for f in $vglogwild ; do
+        case $f in
+    $vglog*) VALGRIND= ;;
+    esac
+      done
+      if test x"$VALGRIND" = x ; then
+    AC_MSG_RESULT([no])
+      else
+    AC_MSG_RESULT([yes])
+    AC_CHECK_HEADERS([valgrind/memcheck.h], [], [VALGRIND=], [ ])
+      fi
+    else
+      dnl The valgrind detected doesn't seem to work!  Perhaps this is an
+      dnl x86_64 box with a 32 bit valgrind.
+      AC_MSG_RESULT([$VALGRIND doesn't work])
+      VALGRIND=
+    fi
+    rm -f $vglogwild
+  fi
+fi
+
+if test -n "$VALGRIND" ; then
+  AC_DEFINE([HAVE_VALGRIND], [1], [Define if a suitable valgrind is installed])
+fi
+
+dnl If eatmydata is installed, we run the testsuite under it to speed it up.
+dnl If EATMYDATA is set to an empty value, then skip this check and don't use
+dnl eatmydata.
+if test -n "${EATMYDATA-unset}" ; then
+  AC_PATH_PROG([EATMYDATA], [eatmydata], [])
+fi
+
+dnl Used by tests/harness/testsuite.cc
+AC_CHECK_FUNCS([sigaction])
+AC_MSG_CHECKING([for sigsetjmp and siglongjmp])
+AC_TRY_COMPILE([#include <setjmp.h>],
+  [sigjmp_buf jb; if (sigsetjmp(jb, 1)) { siglongjmp(jb, 1);  }],
+  AC_DEFINE([HAVE_SIGSETJMP], [1], [Define to 1 if you have the 'sigsetjmp' function])
+  AC_MSG_RESULT([yes]),
+  AC_MSG_RESULT([no]))
+
+dnl ***************************
+dnl * Select modules to build *
+dnl ***************************
+
+dnl Check which database backends should be built.
```

Comment:
:)

Definitely worth at least understanding what the bits are doing, which shouldn't take all that long to get into. Knowing how to write autoconf takes longer, because of things like m4 quoting :-(